remotes/docker/resolver.go

```diff
@@ -295,11 +295,10 @@ func (r *dockerResolver) Resolve(ctx context.Context, ref string) (string, ocisp
 			resp.Body.Close() // don't care about body contents.
 
 			if resp.StatusCode > 299 {
-				if resp.StatusCode == http.StatusNotFound {
-					continue
-				}
-				return "", ocispec.Descriptor{}, errors.Errorf("unexpected status code %v: %v", u, resp.Status)
+				log.G(ctx).Info("trying next host")
```

Comment:
Same here, wondering if the logs should contain more details, e.g.

```suggestion
				log.G(ctx).WithError(errors.Errorf("unexpected status code %v: %v", u, resp.Status)).Info("trying next host")
```

(not sure if "unexpected status code" is correct, as it now also includes `http.StatusNotFound`, which is not "unexpected")

Reply:
Curious (@dmcgowan may know); should `30x` redirects be handled?